src/main/java/org/opensearch/timeseries/util/ParseUtils.java

```diff
@@ -688,6 +696,46 @@ public static String checkFilterByBackendRoles(User requestedUser) {
         return null;
     }
 
+    /**
+     * Verifies whether the user has permission to access the resource.
+     */
+    public static void verifyResourceAccessAndProcessRequest(
+        User requestedUser,
+        String detectorId,
+        NodeClient client,
+        Settings settings,
+        Version nodeVersion,
+        ActionListener<? extends ActionResponse> listener,
+        Consumer<Object[]> onSuccess,
+        Object[] successArgs,
+        BiConsumer<Throwable, Object[]> fallbackOn501,
+        Object[] fallbackArgs
+    ) {
+        ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getResourceSharingClient(client, settings, nodeVersion);
+
+        resourceSharingClient.verifyResourceAccess(detectorId, CommonName.CONFIG_INDEX, ActionListener.wrap(isAuthorized -> {
+            if (!isAuthorized) {
+                listener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            "User " + requestedUser.getName() + " is not authorized to access resource: " + detectorId,
+                            RestStatus.FORBIDDEN
+                        )
+                    );
+                return;
+            }
+            onSuccess.accept(successArgs);
+        }, failure -> {
+            // if resource-sharing feature is not available, proceed with old flow of evaluating filter-by-backend role.
+            if (failure instanceof OpenSearchStatusException
```

Comment:
Can't we just check if the setting is available or not for resource sharing instead of so many exception checks here?

Reply:
We also need to add a TODO or a github issue link to remove the old flow once the resource sharing feature is GA

Reply:
Without these exception checks here, we'd have to define the `resource_sharing` feature flag value inside AD along with a logic to check whether sec plugin is installed. These exception checks allows us to offload the verification to the client, which will send 501 if sec plugin is inactive or resource-sharing feature is disabled.

Reply:
Wouldnt it be possible to expose the feature flag via the SPI?

Reply:
done. consumed the exposed flag.